### PR TITLE
CMake rewrite

### PR DESCRIPTION
**Pull request by Be-ing**
Clean up CMakeLists.txt with modern CMake best practices using targets instead of variables.

This branch is on top of #434.

### CODE REVIEW

**Comment by Be-ing** [comment]
@Holzhaus care to review?

File: CMakeLists.txt [hunk 1]
```diff
@@ -103,384 +16,312 @@ SET(PA_COMMON_SOURCES
...
+  PUBLIC_HEADER "${PORTAUDIO_PUBLIC_HEADERS}"
+  MACOSX_FRAMEWORK_IDENTIFIER com.portaudio
+  FRAMEWORK_VERSION A
+  VERSION ${PROJECT_VERSION}
+  SOVERSION ${PROJECT_VERSION}
+  WINDOWS_EXPORT_ALL_SYMBOLS TRUE
+)
+install(TARGETS portaudio
+  EXPORT portaudio-targets
+  PUBLIC_HEADER DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/portaudio"
```
Comment (Holzhaus) [inline CMakeLists.txt:303]: I think this requires `#include <portaudio/portaudio.h>` instead of `#include <portaudio.h>`.

Comment (Be-ing) [reply CMakeLists.txt]: 3627b03b7ac2b105ae8815f7e5d7aee4ecca67d5

**Comment by Be-ing** [comment]
@philburk did you intend to close a different PR? This is unrelated to f7efb6b.

**Pull request closed without merging**